src/lm/const-arpa-lm.cc

```diff
@@ -29,6 +29,110 @@
 
 namespace kaldi {
 
+// Function for writing array of integer types.
+template<class T> inline void WriteIntegerArray(std::ostream &os, bool binary,
```

Comment:
It occurred to me that this function is not really needed.  Just do the equivalent of
os.write(reinterpret_cast<const char *>(array),      sizeof(T) * array_sz);
and something similar for reading.
of course check os.good() afterward.